ctx.go

```diff
@@ -18,10 +18,20 @@ const (
 	maxBodySize = 1048576
 )
 
-// ctx is the context of the request.
+// ctxWithBody represents the context of the request.
 // It contains the request body, the path parameters, the query parameters, and the HTTP request.
 // Please do not use a pointer type as parameter.
-type ctx[B any] interface {
+type ctxWithBody[B any] interface {
+	CommonCtx[B]
+
+	Context() context.Context
+
+	Request() *http.Request        // Request returns the underlying HTTP request.
+	Response() http.ResponseWriter // Response returns the underlying HTTP response writer.
+}
+
+// ONLY FOR FUEGO INTERNAL USE
+type CommonCtx[B any] interface {
```

Comment:
If it's for internal use, move it to "internal" package, it won't be exported put of your module

Reply:
I'm against internal packages because of circular dependencies and complexity but in this case I think it fits well. 

We need to share internals between `fuego` and `fuegogin` anyway, so I'm not sure there's a better solution anyway.

I'm moving it!